src/app/notification-manager/components/NotificationManager.vue

```diff
@@ -84,8 +85,16 @@ const hasAnyAction = computed(() => {
   }
 })
 
+onMounted(function () {
+  const hideCheckMeshAlert = ClientStorage.get('hideCheckMeshAlert')
+
+  isShowingAlert.value = hideCheckMeshAlert !== 'yes'
+})
+
 function closeAlert(): void {
   isShowingAlert.value = false
+
+  ClientStorage.set('hideCheckMeshAlert', 'yes')
```

Comment:
Oh yeah, true, I must’ve applied that to both PRs while I was looking at it. I’m removing it.

Reply:
Oh, actually, this is different from https://github.com/kumahq/kuma-gui/pull/479#discussion_r1044353036.

It’s such a small change that I’d rather just leave it in here. Its logic is only affecting this file and is of no greater consequence than to persist the dismissal of the "check mesh" alert.

Reply:
https://github.com/kumahq/kuma-gui/pull/466#discussion_r1042000898

> I’m not sure I understand this argument. All this data is runtime state and while it might never change that doesn’t mean it shouldn’t be stored in a Vuex store.

Would you count this as runtime state? You had a very strong opinion that even static, non-changing variables should be stored/retrieved via Vuex. Considering our application is almost exclusively read-only (the source of truth being stored on the backend), and this is possibly one of the very few times we are actually modifying application state, what's your opinion on whether this particular thing should be going via Vuex?

Reply:
Well I dunno, I suppose thats what I'm asking you if that's your point/argument. What's the advantage of not doing it here? This is application state that changes during the application lifecycle, not something that is completely static.

Reply:
Sure, yes, but this data here is not needed anywhere else. It doesn’t need to be _stored_.

Reply:
Fair point, so does that mean:

> I’m not sure I understand this argument. All this data is runtime state and while it might never change that doesn’t mean it shouldn’t be stored in a Vuex store.

Should be more like:

> I’m not sure I understand this argument. All this data is runtime state (apart from when its not needed anywhere else) and while it might never change that doesn’t mean it shouldn’t be stored in a Vuex store.

Reply:
What are your thoughts on using something like Pinia then:

https://pinia.vuejs.org/
> Pinia: The Vue Store that you will enjoy using
> Compared to Vuex, Pinia provides a simpler API with less ceremony, offers Composition-API-style APIs

Looks to me like a lot of Pinia usage within components looks like:

```javascript
import { useSomething, useSomethingElse } from  '@/somewhere'
const something = useSomething();
const somethingElse = useSomethingElse();
```

`something` and `somethingElse` then look and act almost just like regular JS, and they're smaller things, and can be easily and individually be swapped in and out for something different if necessary.

As an aside: I think all in all I'm generally cautious of having this great big single store thing to maintain that we have to keep in sync with the actual source of truth (the backend). It's kinda fine for SPAs that do stuff, like saving/editing and maintaining a source of truth ready to post back somewhere else. It's not like we need to change any extended state/data in our application as yet (or maybe ever?), its just read-only.

I know we can't just rip out existing infra, just wondered what your thoughts were in an ideal world, or even a longer term thing for the app (not saying we need to change anything just good to discuss and hear what you have to say)

Let's chat tomorrow.

Reply:
I'm happy with swapping out Vuex with Pinia. I hope for an overall better typing landscape which is my biggest gripe with Vuex.

Generally, I see a big advantage when using a centralized store: you can add, with relative ease, client-side caching strategies. For example, consider that API calls aren’t directly invoked from our API wrapper (i.e. `kumaApi`) but via a Vuex store action. Such an action could first check for some additional store-kept flag that tells it whether this data was fetched before and opt not to trigger a new, possibly redundant API call.

I’ve implemented something like this before which roughly worked like this:

* a regular store action is available for (1) triggering the API wrapper method and (2) storing its result in the appropriate place
* a lazy variant of that store action is available which triggers the regular variant but only when a condition is fulfilled. In the strategy I implemented, we stored the promise returned by the API wrapper method in the store. This enables the lazy action to check for whether (1) it is set (i.e. the signal to not fetch again) and (2) whether it is fulfilled (used as a way to avoid triggering several parallel asynchronous fetches).

On top of this approach, one could add a timer that clears the stored signal (e.g. the promise) after a certain while which would be the cache busting part. Of course certain pages/actions would just simply trigger the non-lazy action to avoid this caching (e.g. the reload buttons we have in some places).

---

The synchronization of data question to me is only partially related to the question of whether one has a central store like a Vuex store. Either with or without such a store, one needs to implement a strategy for ensuring the data is as accurate as it needs to do while simultaneously trying to avoid to constantly fetch data which would put unnecessary load on the API server.

Reply:
I think I mostly agree with the majority of this 👍 , but obviously for things that change and that we should keep listening (or checking) for changes to.